### PR TITLE
Cleanup UnicodeEncoding tests

### PR DESCRIPTION
**Pull request by hughbe**
- Rename tests to follow guidelines, instead of "PosTest1" and
"NegTest2"
- Take advantage of xunit
- Shorten tests by using quicker and newer code

In this commit, I also factor out some common test code amongst all encoding classes. Currently, only UnicodeEncoding uses this, but if/when this PR is merged, I will use this class (EncodingHelpers) in other tests.

This approach ensures consistent implementations across all encodings

### CODE REVIEW

File: src/System.Text.Encoding/tests/EncodingTestHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            Assert.Throws<ArgumentOutOfRangeException>("bytes", () => encoding.GetString(new byte[1], 2, 0));
+            Assert.Throws<ArgumentOutOfRangeException>("bytes", () => encoding.GetString(new byte[1], 1, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("bytes", () => encoding.GetString(new byte[1], 0, 2));
+        }
+
+        public static string GetString() => GetString(2, 260);
+        public static string GetString(int length) => GetString(length, length);
+        public static string GetString(int minLength, int maxLength) => s_randomGenerator.GetString(-55, false, minLength, maxLength);
+
+        public static string GetUnicodeString(int length)
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: Should this just be GetRandomString?

**Comment by stephentoub** [comment]
LGTM

cc: @tarekgh

File: src/System.Text.Encoding/tests/EncodingTestHelpers.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: I'm confused as to what you mean here. Do you mean rename this helper method to "GetRandomString", or replace all usage of this metod with `GetString(int length)` from above?

Comment (stephentoub) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: I simply meant to rename the helper. Strings are UTF16, so "Unicode" here doesn't seem to be adding much, whereas "Random" is more descriptive of what it's doing.

**Comment by tarekgh** [comment]
LGTM too.

File: src/System.Text.Encoding/tests/EncodingTestHelpers.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: Sure thing. Fixed

**Comment by stephentoub** [comment]
Test Innerloop OSX Release Build and Test please (restore hang)

**Comment by stephentoub** [comment]
Test Innerloop CentOS7.1 Release Build and Test  please

@mmitche, what causes an error like this?
```
04:45:32 Ignoring paths matching patterns: ["netci.groovy"]
04:45:32 Empty changeset, running build.
06:45:32 
Build timed out (after 120 minutes). Marking the build as aborted.
06:45:32 FATAL: Failed to copy /jenkins/jobs/dotnet_coreclr/jobs/master/jobs/release_centos7.1/builds/1/archive/bin/Product/Linux.x64.Release/libmscordaccore.so to /mnt/resource/j/workspace/dotnet_corefx/centos7.1_release_tst_prtest/bin/Product/Linux.x64.Release/libmscordaccore.so
06:45:32 hudson.util.IOException2: Failed to copy 
```
This is the second or third time I've seen this in the last day or so.

**Pull request merged**